src/categories/categories.controller.ts

```diff
@@ -0,0 +1,66 @@
+import {
+  Body,
+  Controller,
+  Delete,
+  Get,
+  Param,
+  Patch,
+  Post,
+} from '@nestjs/common';
+import { ApiOperation, ApiParam, ApiResponse, ApiTags } from '@nestjs/swagger';
+import { Category } from './categories.model';
+import { CategoryService } from './categories.service';
+import { CreateCategoryDto } from './dto/create-category.dto';
+
+@ApiTags('Category')
+@Controller('category')
+export class CategoryController {
+  constructor(private categoryService: CategoryService) {}
+
+  @ApiOperation({ summary: 'Get Category list' })
+  @ApiResponse({ status: 200, type: [Category] })
+  @Get()
+  getAll() {
+    return this.categoryService.getAllCategory();
+  }
+
+  @ApiOperation({ summary: 'Create category' })
+  @ApiResponse({ status: 200, type: Category })
+  @Post()
+  create(@Body() categoryDto: CreateCategoryDto): Promise<Category> {
+    return this.categoryService.createCategory(categoryDto);
+  }
+
+  @ApiOperation({
+    summary: 'category by id',
+  })
+  @ApiResponse({ status: 200, type: Category })
+  @Get('/:id')
+  @ApiParam({ name: 'id' })
+  getProduct(@Param() param: { id: string }): any {
+    return this.categoryService.getCategooryById(param.id);
+  }
+
+  @ApiOperation({
+    summary: 'delete category',
+  })
+  @ApiResponse({ status: 204 })
+  @Delete('/:id')
+  @ApiParam({ name: 'id' })
+  deleteProduct(@Param() param: { id: string }): any {
```

Comment:
Dont use any as a return type. Sequelize has own types for delete/update results.